bitbucket-slack-server-integration-plugin/src/main/java/com/atlassian/bitbucket/plugins/slack/listener/BitbucketNotificationEventListener.java

```diff
@@ -101,11 +102,26 @@ public void onEvent(final PullRequestEvent event) {
                             getUsersAddedToPullRequest(event),
                             false),
                     options -> {
+                        final boolean isReviewersUpdate = (event instanceof PullRequestOpenedEvent && !event.getPullRequest().getReviewers().isEmpty()) ||
```

Comment:
Is there any specific reason why the event processing logic was moved here inside the callback from he findChannelsAndPublishNotificationsAsync method?
I ask because this callback might be called multiple times, for each notifications, so anything that does not depend on the `options` argument can be moved above, out of this callback, and thus it will be processed only once.
It's not super critical as we're not doing anything heavyweight here but it's still good practice.